comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID       = "local-process-collector"
+	cacheValidityNoRT = 2 * time.Second
+	componentName     = "workloadmeta-process"
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+}
+
+// NewCollector returns a new docker collector provider and an error
+func NewCollector() (workloadmeta.CollectorProvider, error) {
+	return workloadmeta.CollectorProvider{
+		Collector: &collector{
+			id:      collectorID,
+			catalog: workloadmeta.NodeAgent,
+		},
+	}, nil
+}
+
+// GetFxOptions returns the FX framework options for the collector
+func GetFxOptions() fx.Option {
+	return fx.Provide(NewCollector)
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Component) error {
+	if !config.Datadog().GetBool("language_detection.enabled") && flavor.GetFlavor() != flavor.DefaultAgent {
+		return errors.NewDisabled(componentName, "core agent language detection not enabled")
+	}
+	c.store = store
+	c.processDiffCh = processwlm.GetSharedWorkloadMetaExtractor(config.SystemProbe).ProcessCacheDiff()
+
+	go c.stream(ctx)
+
+	return nil
+}
+
+func CollectorEventsFromProcessDiff(creations []*processwlm.ProcessEntity, deletions []*processwlm.ProcessEntity) []workloadmeta.CollectorEvent {
+	events := make([]workloadmeta.CollectorEvent, 0, len(creations)+len(deletions))
+
+	for _, creation := range creations {
+		events = append(events, workloadmeta.CollectorEvent{
+			Type: workloadmeta.EventTypeSet,
+			Entity: &workloadmeta.Process{
+				EntityID: workloadmeta.EntityID{
+					Kind: workloadmeta.KindProcess,
+					ID:   strconv.Itoa(int(creation.Pid)),
+				},
+				ContainerID:  creation.ContainerId,
+				NsPid:        creation.NsPid,
+				CreationTime: time.UnixMilli(creation.CreationTime),
+				Language:     creation.Language,
+			},
+			Source: workloadmeta.SourceLocalProcessCollector,
+		})
+	}
+
+	for _, deletion := range deletions {
+		events = append(events, workloadmeta.CollectorEvent{
+			Type: workloadmeta.EventTypeUnset,
+			Entity: &workloadmeta.Process{
+				EntityID: workloadmeta.EntityID{
+					Kind: workloadmeta.KindProcess,
+					ID:   strconv.Itoa(int(deletion.Pid)),
+				},
+			},
+			Source: workloadmeta.SourceLocalProcessCollector,
+		})
+	}
+
+	return events
+}
+
+func (c *collector) stream(ctx context.Context) error {
+	ctx, cancel := context.WithCancel(ctx)
+	healthHandle := health.RegisterLiveness(componentName)
+	for {
+		select {
+		case <-ctx.Done():
+			err := healthHandle.Deregister()
+			if err != nil {
+				log.Warnf("error de-registering health check: %s", err)
+			}
+			cancel()
+			return nil
```

Comment:
no need to return `nil`. 

A simple `return` is enough to stop the go routine.

Reply:
Done